extensions/VisualEditor/wikia/ApiAddMediaTemporary.php

```diff
@@ -154,6 +156,14 @@ protected function checkPermissions() {
 		}
 	}
 
+	protected function checkPermissionsForJA() {
```

Comment:
Well, let's put performance aside for a second.
If you can write logic based on generic variables such as wgdisableanonymousediting then I think you should - it's cleaner and cheaper to maintain than adding more of a lang specific logic.